include/ifc/abstract-sgraph.hxx

```diff
@@ -3544,7 +3544,18 @@ namespace ifc {
 
     // -- exception type in case of an invalid partition name
     struct InvalidPartitionName {
-        std::string_view name;
+        static constexpr unsigned NAME_BUFFER_SIZE = 64;
+
+        char name[NAME_BUFFER_SIZE]                = "";
+
+        static InvalidPartitionName make(std::string_view name) noexcept
```

Comment:
Why a factory function instead of an explicit ctor? Copying from one buffer into another can't fail, so there is no reason to make this a factory function.

Reply:
In one of my previous attempts (https://github.com/microsoft/ifc/pull/30#discussion_r1344658604), @cdacamar told me not to introduce a ctor in this class (but maybe his the rationale does not apply if we have some real work to do anyway in the ctor)

Reply:
The comment was on an implicit converting ctor that did the same thing the compiler generated one would have, so that comment was valid. However, here the work is non-trivial, so this can be a ctor.